x/group/client/testutil/tx.go

```diff
@@ -2347,3 +2554,152 @@ func (s *IntegrationTestSuite) createCLIProposal(groupPolicyAddress, proposer, s
+	clientCtx := val.ClientCtx
+	commonFlags := []string{
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+	}
+
+	membersValid := validMembers{}
+
+	for i, address := range membersAddress {
+		membersValid.Members = append(membersValid.Members, member{
+			address,
+			membersWeight[i],
+			validMetadata,
+		})
+	}
+	validMemberByte, err := json.MarshalIndent(&membersValid, "", "\t")
+	s.Require().NoError(err)
+	validMemberString := string(validMemberByte)
+	s.Require().NoError(err)
+	validMembersFile := testutil.WriteToNewTempFile(s.T(), validMemberString)
+	out, err := cli.ExecTestCLICmd(clientCtx, client.MsgCreateGroupCmd(),
+		append(
+			[]string{
+				val.Address.String(),
+				validMetadata,
+				validMembersFile.Name(),
+			},
+			commonFlags...,
+		),
+	)
+	s.Require().NoError(err, out.String())
+	var txResp sdk.TxResponse
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txResp), out.String())
+	return s.getGroupIdFromTxResponse(txResp)
+}
+
+func (s *IntegrationTestSuite) createGroupThresholdPolicyWithDenom(groupID string, threshold int, denon int64) string {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+
+	commonFlags := []string{
+		fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+		fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+	}
+
+	out, err := cli.ExecTestCLICmd(clientCtx, client.MsgCreateGroupPolicyCmd(),
+		append(
+			[]string{
+				val.Address.String(),
+				groupID,
+				validMetadata,
+				fmt.Sprintf("{\"@type\":\"/cosmos.group.v1.ThresholdDecisionPolicy\", \"threshold\":\"%d\", \"windows\":{\"voting_period\":\"30000s\"}}", threshold),
+			},
+			commonFlags...,
+		),
+	)
+	var txResp = sdk.TxResponse{}
+	s.Require().NoError(err, out.String())
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &txResp), out.String())
+	s.Require().Equal(uint32(0), txResp.Code, out.String())
+
+	out, err = cli.ExecTestCLICmd(val.ClientCtx, client.QueryGroupPoliciesByGroupCmd(), []string{groupID, fmt.Sprintf("--%s=json", tmcli.OutputFlag)})
+	s.Require().NoError(err, out.String())
+	var res group.QueryGroupPoliciesByGroupResponse
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &res))
+	groupPolicyAddress := res.GroupPolicies[0].Address
+
+	addr, err := sdk.AccAddressFromBech32(groupPolicyAddress)
+	s.Require().NoError(err)
+	out, err = banktestutil.MsgSendExec(clientCtx, val.Address, addr,
+		sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(denon))),
+		commonFlags...,
+	)
+	s.Require().NoError(err)
+	return groupPolicyAddress
+}
+
+func (s *IntegrationTestSuite) newValidMembers(weights, membersAddress []string) validMembers {
```

Comment:
I'm not sure I understand the purpose of this new function (and of all the new types defined below).

Could you elaborate on this?

Reply:
The only purpose is to generate member files for the policy, they are structs so I can easily serialize them into a string. Let me know if you rather I change it to something different.

Reply:
Ah, understood. So in the SDK we never (or rarely) use `json.MarshalIndent` from the std json package. Instead, we use ProtoJSON, see `cdc.MarshalJSON`.

I think we can keep this function, but we don't need the new types below, and we can probably use the existing types with our `cdc` codec.

Reply:
fixed